src/relay/transforms/to_mixed_precision.cc

```diff
@@ -381,6 +400,11 @@ class MixedPrecisionPass : public MixedModeMutator {
       if (accumulation_dtype != output_dtype) {
         output = CastArg(output, GetType(output), output_dtype);
       }
+      if (pre_call_node == static_cast<const CallNode*>(root_) && enable_original_type_) {
```

Comment:
Could we just use `.same_as`?

Reply:
changing to same_as will require us to use "if (((GetRef<Call>(pre_call_node)).same_as(GetRef<RelayExpr>(root_))) && keep_orig_output_dtype_ )". So leaving the comparison as  such